includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -439,6 +439,7 @@ public function init_hooks() {
 			add_action( 'woocommerce_update_order', [ $this, 'schedule_order_tracking' ], 10, 2 );
 
 			add_filter( 'rest_request_before_callbacks', [ $this, 'remove_all_actions_on_preflight_check' ], 10, 3 );
+			add_action( 'woocommerce_email_before_order_table', [ $this, 'set_payment_method_title_for_email' ], 10, 3 );
```

Comment:
I'd maybe vote towards moving this into the [`init_hooks()`](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-upe-checkout.php#L81) function inside `WC_Payments_UPE_Checkout`. This hook is only intended to be fired once and only required for the UPE. Putting it here inside `WC_Payment_Gateway_WCPay` satisfies the first goal, but this hook will also be added when the legacy card is enabled and UPE is disabled after [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments.php#L542).

Attaching this hook with UPE disabled probably won't break anything and likely won't be a long-term issue, as we move away from the gateway, but seems redundant nonetheless. 🤷

Reply:
Agree with the fact that although there is no negative impact while initialized in the `WC_Payment_Gateway_WCPay`, still `WC_Payments_UPE_Checkout` seems to be a more appropriate place for it. Done in https://github.com/Automattic/woocommerce-payments/pull/6826/commits/11ac6ed5014ee3bf6efffa8d5b20223df7d2a96a